### PR TITLE
Move Build cancellation and Pod deletion logic to HandleBuild

### LINKED ISSUE #3353: Cancel build... doesn't
Cancelling a running build doesn't work very well.  We should be able to kill the pod almost immediately.

### PR DESCRIPTION
**Pull request by mnagy**
This will delete the pod right away. Fixes #3353.

cc @bparees

### CODE REVIEW

**Comment by bparees** [comment]
please move cancelbuild back where it was so the diff will be cleaner.

**Comment by mnagy** [comment]
I moved it because it now doesn't have BuildPodController as receiver, but BuildController. If it'd been there, it would be between BuildPodController methods.

File: pkg/build/controller/controller.go [hunk 1]
```diff
@@ -171,18 +209,7 @@ func (bc *BuildPodController) HandlePod(pod *kapi.Pod) error {
 
 	build := obj.(*buildapi.Build)
 
-	// A cancelling event was triggered for the build, delete its pod and update build status.
-	if build.Status.Cancelled {
-		glog.V(4).Infof("Cancelling Build %s/%s.", build.Namespace, build.Name)
-
-		if err := bc.CancelBuild(build, pod); err != nil {
-			return fmt.Errorf("failed to cancel Build %s/%s: %v, will retry", build.Namespace, build.Name, err)
```
Comment (bparees) [inline pkg/build/controller/controller.go]: why didn't you move this message/logic up into handlebuild?

File: pkg/build/controller/controller.go [hunk 2]
```diff
@@ -266,6 +269,11 @@ func (bc *BuildPodDeleteController) HandleBuildPodDeletion(pod *kapi.Pod) error
 	}
 	build := obj.(*buildapi.Build)
 
+	// If build was cancelled, we'll leave HandleBuild to update the build
+	if build.Status.Cancelled {
+		return nil
```
Comment (bparees) [inline pkg/build/controller/controller.go]: lov.V(4) message here, like below so we know why it didn't do anything.

File: pkg/build/controller/controller.go [hunk 3]
```diff
@@ -41,12 +41,50 @@ type imageStreamClient interface {
...
+
+	build.Status.Phase = buildapi.BuildPhaseCancelled
+	now := util.Now()
+	build.Status.CompletionTimestamp = &now
+	if err := bc.BuildUpdater.Update(build.Namespace, build); err != nil {
+		return err
+	}
+
+	glog.V(4).Infof("Build %s/%s was successfully cancelled.", build.Namespace, build.Name)
+	glog.Infof("Build %s/%s was successfully cancelled.", build.Namespace, build.Name)
```
Comment (mnagy) [inline pkg/build/controller/controller.go]: Seems like I left double logs here from testing.

**Comment by mnagy** [comment]
@bparees PTAL. Addressed comments and fixed tests.

**Comment by bparees** [comment]
lgtm, needs a squash

**Comment by mnagy** [comment]
Done.

**Comment by mnagy** [comment]
Hmm, not sure why tests fail, but seems unrelated.

**Comment by bparees** [comment]
merge queue had issues.  [merge]

**Pull request merged**